src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -440,6 +442,21 @@ int sd_dhcp_client_set_vendor_class_identifier(
         return free_and_strdup(&client->vendor_class_identifier, vci);
 }
 
+int sd_dhcp_client_set_user_class(
+                sd_dhcp_client *client,
+                const char **user_class) {
+
+        assert_return(client, -EINVAL);
+
+        client->user_class = strv_free(client->user_class);
+
+        client->user_class = strv_copy((char **) user_class);
+        if (!client->user_class)
+                return -ENOMEM;
```

Comment:
not that it matetrs much but i think it would be nice to make this call "atomic", i.e. that when it fals the client object is not modified. Specifically this means copying the list first, storing it into some temporary variable, and only then freeing the old version

Reply:
yes that nice. Thanks !